Thanks to whoever enabled the CI. :+1: 

The macOS build tests are failing but given

https://github.com/svalinn/DAGMC/blob/a435d0647f3e3e19f9161a42b4bbbdb4be65bf65/.github/workflows/mac_build_test.yml#L58-L75

and 

```
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for memory.h
-- Looking for memory.h - found
-- Check if the system is big endian
-- Searching 16 bit integer
-- Check size of unsigned short
CMake Error: File /opt/homebrew/share/cmake/Modules/TestEndianess.c.in does not exist.
-- Check size of unsigned short - done
CMake Error at config/TestBigEndian.cmake:39 (CONFIGURE_FILE):
-- Using unsigned short
  CONFIGURE_FILE Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:168 (TEST_BIG_ENDIAN)


CMake Error at config/TestBigEndian.cmake:43 (FILE):
  FILE failed to open for reading (No such file or directory):

    /Users/runner/work/DAGMC/moab/bld/CMakeFiles/CMakeTmp/TestEndianess.c
-- Configuring incomplete, errors occurred!
Call Stack (most recent call first):
  CMakeLists.txt:168 (TEST_BIG_ENDIAN)


CMake Error at /Users/runner/work/DAGMC/moab/bld/CMakeFiles/CMakeTmp/CMakeLists.txt:21 (target_sources):
  Cannot find source file:

    /Users/runner/work/DAGMC/moab/bld/CMakeFiles/CMakeTmp/TestEndianess.c


CMake Error at /Users/runner/work/DAGMC/moab/bld/CMakeFiles/CMakeTmp/CMakeLists.txt:20 (add_executable):
  No SOURCES given to target: cmTC_2e25d


CMake Error at config/TestBigEndian.cmake:46 (TRY_COMPILE):
  Failed to generate test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:168 (TEST_BIG_ENDIAN)


make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target `install'.  Stop.
```

this seems to be unrelated to this PR specifically and more issues with what's currently installed by the Homebrew `cmake` on the runner(?).

--- a/.github/workflows/mac_build_test.yml
+++ b/.github/workflows/mac_build_test.yml
@@ -16,7 +16,6 @@ on:
       - '.github/actions/**'
       - 'CI/**'
       - 'doc/**'
-
   push:
     branches:
       - develop
@@ -47,13 +46,24 @@ jobs:
       - name: Initial setup
         shell: bash -l {0}
         run: |
-          brew install eigen hdf5
+          brew install hdf5
+
+          # Homebrew's Eigen in 2026 is Eigen v5.x, which is incompatible with
+          # MOAB v5.5.1 FindEigen3.cmake. Homebrew has bad version control, and
+          # so instead of trying to force Homebrew to install a compatible version,
+          # build Eigen v3.4.0 (a header-only library) from source to ensure that
+          # Eigen3Config.cmake is generated for DAGMC's CMake invocation to use
+          # find_package(Eigen3 REQUIRED NO_MODULE).
+          curl -sL https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.bz2 | tar xj -C /tmp
+          cmake -DCMAKE_INSTALL_PREFIX=/tmp/eigen -S /tmp/eigen-3.4.0 -B /tmp/eigen-build
+          cmake --build /tmp/eigen-build --parallel
+          cmake --install /tmp/eigen-build
+          echo "EIGEN_PREFIX=/tmp/eigen" >> $GITHUB_ENV
 
       - name: Environment Variables
         shell: bash -l {0}
         run: |
           echo "HOME=$GITHUB_WORKSPACE/.." >> $GITHUB_ENV
-
 
       - name: Build MOAB
         shell: bash -l {0}
@@ -62,6 +72,23 @@ jobs:
           mkdir -pv moab/bld
           cd moab
           git clone https://bitbucket.org/fathomteam/moab -b 5.5.1 --depth 1 --shallow-submodules
+
+          # In CMavke v4 TestEndianess.c.in was renamed to TestEndianness.c.in
+          # to fix a typo. MOAB v5.5.1 used a custom config/TestBigEndian.cmake
+          # that hardcodes the use of TestEndianess.c.in.
+          #
+          #  $ git grep "TestEndianess.c.in"
+          #  config/TestBigEndian.cmake:39:    CONFIGURE_FILE("${CMAKE_ROOT}/Modules/TestEndianess.c.in"
+          #
+          # As config/ is first in CMAKE_MODULE_PATH, MOAB's broken version is
+          # used instead of CMake's working built-in module. To avoid this, set
+          # HAVE_WORDS_BIGENDIAN=TRUE to skip the test, instead of deleting the
+          # broken module. This works as HAVE_WORDS_BIGENDIAN=TRUE skips the test
+          # test_big_endian, defaulting to little-endian, which for effectively all
+          # runner platforms is correct.
+          #
+          # $ git grep -m 1 '(HAVE_' -- config/TestBigEndian.cmake
+          # config/TestBigEndian.cmake:8:  IF(HAVE_${VARIABLE} MATCHES ^HAVE_${VARIABLE}$)
           cd bld
           cmake ../moab -DENABLE_HDF5=ON \
                         -DENABLE_PYMOAB=OFF \
@@ -69,6 +96,9 @@ jobs:
                         -DBUILD_SHARED_LIBS=ON \
                         -DENABLE_BLASLAPACK=OFF \
                         -DENABLE_FORTRAN=OFF \
+                        -DHAVE_WORDS_BIGENDIAN=TRUE \
+                        -DCMAKE_PREFIX_PATH="$(brew --prefix)" \
+                        -DEIGEN3_DIR="${EIGEN_PREFIX}/include/eigen3"
                         -DCMAKE_INSTALL_PREFIX=${HOME}/moab
           make
           make install
@@ -80,6 +110,7 @@ jobs:
           mkdir -p $GITHUB_WORKSPACE/bld
           cd $GITHUB_WORKSPACE/bld
           cmake ../ -DMOAB_DIR=${HOME}/moab \
+                    -DCMAKE_PREFIX_PATH="${EIGEN_PREFIX};$(brew --prefix)" \
                     -DBUILD_CI_TESTS=ON \
                     -DBUILD_STATIC_EXE=OFF \
                     -DBUILD_SHARED_LIBS=ON \

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
+cmake_minimum_required(VERSION 3.18...4.2)
 project(DAGMC)
-cmake_minimum_required(VERSION 3.18)
 enable_language(CXX)
 
 # Set DAGMC version

--- a/cmake/test_config/CMakeLists.txt
+++ b/cmake/test_config/CMakeLists.txt
@@ -1,5 +1,5 @@
+cmake_minimum_required(VERSION 2.8...4.2)
 project(Test_DAGMC_config)
-cmake_minimum_required(VERSION 2.8)
 
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR})
 

--- a/doc/CHANGELOG.rst
+++ b/doc/CHANGELOG.rst
@@ -16,6 +16,7 @@ Next version
 
 **Fixed**
   * Fixed HDF5 naming convention for docker container building and naming (#976)
+  * Allowed for modern CMake policies to be used (#989)
 
 
 v3.2.4

--- a/src/geant4/generate_geant4
+++ b/src/geant4/generate_geant4
@@ -1296,7 +1296,7 @@ def _setup_build(dag_h5m):
 
     file_contents.append('#----------------------------------------------------------------------------\n')
     file_contents.append('# Setup the project\n')
-    file_contents.append('cmake_minimum_required(VERSION 2.6 FATAL_ERROR)\n')
+    file_contents.append('cmake_minimum_required(VERSION 2.6...4.2)\n')
     file_contents.append('project(N01)\n')
     file_contents.append('\n')
     file_contents.append('#----------------------------------------------------------------------------\n')

--- a/src/gtest/gtest-1.8.0/CMakeLists.txt
+++ b/src/gtest/gtest-1.8.0/CMakeLists.txt
@@ -26,8 +26,8 @@ endif()
 # as ${gtest_SOURCE_DIR} and to the root binary directory as
 # ${gtest_BINARY_DIR}.
 # Language "C" is required for find_package(Threads).
+cmake_minimum_required(VERSION 2.6...4.2)
 project(gtest CXX C)
-cmake_minimum_required(VERSION 2.6.4)
 
 # Where gtest's .h files can be found.
 include_directories(

